app/src/main/java/dev/animeshvarma/sigil/util/SigilPreferences.kt

```
@@ -91,4 +105,159 @@ class SigilPreferences(context: Context) {
+
+                    profiles.add(EncryptionProfile(
+                        id = id,
+                        name = name,
+                        description = desc,
+                        layers = layers,
+                        kdfConfig = kdfConfig,
+                        isBuiltIn = false,
+                        isCompressionEnabled = compress,
+                        isRaw = isRaw
+                    ))
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)